solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1307,16 +1306,17 @@ public String register(
         boolean joinAtHead = replica.getBool(SliceMutator.PREFERRED_LEADER_PROP, false);
         if (replica.getType().leaderEligible) {
           joinElection(desc, afterExpiration, joinAtHead);
-        } else if (replica.getType() == Type.PULL) {
+        } else {
           if (joinAtHead) {
             log.warn(
-                "Replica {} was designated as preferred leader but it's type is {}, It won't join election",
+                "Replica {} was designated as preferred leader but its type is {}, It won't join election",
                 coreZkNodeName,
-                Type.PULL);
+                replica.getType());
           }
           log.debug(
-              "Replica {} skipping election because it's type is {}", coreZkNodeName, Type.PULL);
-          startReplicationFromLeader(coreName, false);
```

Comment:
Weirdly, we were starting the replication for PULL replicas here, and lower for TLOG replicas. We were also starting unconditionally the replication for PULL replicas, even if we were entering in recovery mode (where on of the first actions is to stop the replication). Replication is now started for both replica types at the same place.